mindtrace/database/mindtrace/database/backends/redis_odm_backend.py

```diff
@@ -325,3 +325,135 @@ def get_raw_model(self) -> Type[ModelType]:
                 print(f"Using model: {model_class.__name__}")
         """
         return self.model_cls
+
+    # Asynchronous wrapper methods for compatibility
+    async def initialize_async(self):
+        """
+        Initialize the Redis connection asynchronously (wrapper around sync initialize).
+
+        This method provides an asynchronous interface to the sync initialize method.
+        It should be called before performing any database operations in an async context.
+
+        Example:
+            .. code-block:: python
+
+                backend = RedisMindtraceODMBackend(User, "redis://localhost:6379")
+                await backend.initialize_async()  # Can be called from async code
+        """
+        # For sync operations, we can just call them directly
+        # They're fast enough that blocking the event loop briefly is acceptable
+        self.initialize()
+
+    async def insert_async(self, obj: BaseModel) -> ModelType:
+        """
+        Insert a new document asynchronously (wrapper around sync insert).
+
+        Args:
+            obj (BaseModel): The document object to insert into the database.
+
+        Returns:
+            ModelType: The inserted document with generated fields populated.
+
+        Raises:
+            DuplicateInsertError: If the document violates unique constraints.
+
+        Example:
+            .. code-block:: python
+
+                user = User(name="John", email="john@example.com")
+                try:
+                    inserted_user = await backend.insert_async(user)
+                    print(f"Inserted user with ID: {inserted_user.pk}")
+                except DuplicateInsertError as e:
+                    print(f"Duplicate entry: {e}")
+        """
+        # Call sync method directly - Redis operations are fast
+        return self.insert(obj)
```

Comment:
These direct calls will still block the event loop. Similarly for other ops (initialize/get/find/delete/all_async)

```suggestion
        return await asyncio.to_thread(self.insert, obj)
```